js/__internal/grids/grid_core/search/m_search.ts

```diff
@@ -232,8 +233,10 @@ export const searchModule = {
 
         _getStringNormalizer() {
           const isCaseSensitive = this.option('searchPanel.highlightCaseSensitive');
-          return function (str) {
-            return isCaseSensitive ? str : str.toLowerCase();
+          const langParams = this.getController('data')?.getDataSource?.()?.loadOptions?.()?.langParams;
+
+          return function (str: string): string {
+            return toComparable(str, isCaseSensitive, langParams);
           };
```

Comment:
Can we return the arrow function here?

```suggestion
          return (str: string): string => toComparable(str, isCaseSensitive, langParams);
```